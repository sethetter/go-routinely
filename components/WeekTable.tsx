import * as React from 'react'

import {
  weekDaysFromDate
} from '../lib/helpers'

import ActivityRow from './ActivityRow'
import WeekTableHead from './WeekTableHead'

interface WeekTableProps { 
  startOfWeek: Date
  activities: Activity[]
  logsForWeek: ActivityLog[]
}

const WeekTable = (props: WeekTableProps) => {
  const { startOfWeek, activities, logsForWeek } = props
  const weekDays = weekDaysFromDate(startOfWeek)
  
  const activityRows = activities.map(a => {
    const logsForActivity = logsForWeek.filter(l => l.activityId === a.id)
    return (
      <ActivityRow
        key={a.id}
        activity={a}
        weekDays={weekDays}
        logsForActivity={logsForActivity}
      />
    )
  })

  return (
    <table className="table table-bordered WeekTable">
      <WeekTableHead startOfWeek={startOfWeek} />
      <tbody>{activityRows}</tbody>
    </table>
  )
}

export default WeekTable
