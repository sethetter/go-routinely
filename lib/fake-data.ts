import * as moment from 'moment'

interface AppData {
  startOfWeek: Date
  activities: Activity[]
  activityLogs: ActivityLog[]
  weekDays: Date[]
}

const startOfWeek = new Date('03/25/2018')

const WEEKDAYS = [
  moment(startOfWeek).day('Sunday').startOf('day').toDate(),
  moment(startOfWeek).day('Monday').startOf('day').toDate(),
  moment(startOfWeek).day('Tuesday').startOf('day').toDate(),
  moment(startOfWeek).day('Wednesday').startOf('day').toDate(),
  moment(startOfWeek).day('Thursday').startOf('day').toDate(),
  moment(startOfWeek).day('Friday').startOf('day').toDate(),
  moment(startOfWeek).day('Saturday').startOf('day').toDate(),
]

const DATA: AppData = {
  startOfWeek,
  activities: [
    { id: 1, name: 'Brush teeth', value: 1 },
    { id: 2, name: 'Taekwondo', value: 2 },
    { id: 3, name: 'Clean room', value: 3 },
  ],
  activityLogs: [
    { id: 1, name: 'Brush teeth', value: 1, activityId: 1, occurredAt: WEEKDAYS[0] },
    { id: 2, name: 'Brush teeth', value: 1, activityId: 1, occurredAt: WEEKDAYS[0] },
    { id: 3, name: 'Taekwondo', value: 2, activityId: 2, occurredAt: WEEKDAYS[3] },
    { id: 4, name: 'Clean room', value: 3, activityId: 3, occurredAt: WEEKDAYS[2] },
  ],
  weekDays: WEEKDAYS
}

export const getFakeData = (): AppData => Object.assign({}, DATA)