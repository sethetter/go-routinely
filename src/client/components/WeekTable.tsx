import * as React from 'react'

import {
  weekDaysFromDate
} from '../lib/helpers'

import ActivityRow from './ActivityRow'
import WeekTableHead from './WeekTableHead'

interface WeekTableProps { 
  startOfWeek: Date
  activities: Activity[]
  logsForWeek: ActivityLog[]
  createLog: (params: Partial<ActivityLog>) => void
}

const WeekTable = (props: WeekTableProps) => {
  const { startOfWeek, activities, logsForWeek, createLog } = props
  const weekDays = weekDaysFromDate(startOfWeek)
  
  const activityRows = activities.map(a => {
    const logsForActivity = logsForWeek.filter(l => l.activityId === a._id)
    return (
      <ActivityRow
        key={a._id}
        activity={a}
        weekDays={weekDays}
        logsForActivity={logsForActivity}
        createLog={createLog}
      />
    )
  })

  return (
    <table className="table table-bordered WeekTable">
      <WeekTableHead startOfWeek={startOfWeek} />
      <tbody>{activityRows}</tbody>
    </table>
  )
}

export default WeekTable
