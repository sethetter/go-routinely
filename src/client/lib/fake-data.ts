import moment from 'moment'

interface AppData {
  user: UserData
  startOfWeek: Date
  activities: Activity[]
  activityLogs: ActivityLog[]
  weekDays: Date[]
}

const startOfWeek = new Date('03/25/2018')

const WEEKDAYS = [
  moment(startOfWeek).day('Sunday').startOf('day').toDate(),
  moment(startOfWeek).day('Monday').startOf('day').toDate(),
  moment(startOfWeek).day('Tuesday').startOf('day').toDate(),
  moment(startOfWeek).day('Wednesday').startOf('day').toDate(),
  moment(startOfWeek).day('Thursday').startOf('day').toDate(),
  moment(startOfWeek).day('Friday').startOf('day').toDate(),
  moment(startOfWeek).day('Saturday').startOf('day').toDate(),
]

const DATA: AppData = {
  startOfWeek,
  user: { id: 'aosidfjoasidfj' },
  activities: [
    { _id: '1', name: 'Brush teeth', value: 1 },
    { _id: '2', name: 'Taekwondo', value: 2 },
    { _id: '3', name: 'Clean room', value: 3 },
  ],
  activityLogs: [
    { _id: '1', name: 'Brush teeth', value: 1, activityId: '1', occurredAt: WEEKDAYS[0] },
    { _id: '2', name: 'Brush teeth', value: 1, activityId: '1', occurredAt: WEEKDAYS[0] },
    { _id: '3', name: 'Taekwondo', value: 2, activityId: '2', occurredAt: WEEKDAYS[3] },
    { _id: '4', name: 'Clean room', value: 3, activityId: '3', occurredAt: WEEKDAYS[2] },
  ],
  weekDays: WEEKDAYS
}

export const getFakeData = (): AppData => Object.assign({}, DATA)
