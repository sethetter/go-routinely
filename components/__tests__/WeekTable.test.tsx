import * as React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import { getFakeData } from '../fake-data'

import WeekTable from './WeekTable'

const DATA = getFakeData()

describe('<WeekTable />', () => {
  it('should render correctly', () => {
    const component = shallow(
      <WeekTable
        isFetching={false}
        weekDays={DATA.weekDays}
        activities={DATA.activities}
        logsForWeek={DATA.activityLogs}
      />
    )

    expect(shallowToJson(component)).toMatchSnapshot()
  })

  // it('shows "loading" when fetching', () => {})
  // it('renders a row for each activity', () => {})
  // it('filters logs for each activity', () => {})
})
