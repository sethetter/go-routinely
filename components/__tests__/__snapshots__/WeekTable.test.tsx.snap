// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeekTable /> should render correctly 1`] = `
<table
  className="table table-bordered WeekTable"
>
  <WeekTableHead
    startOfWeek={2018-04-08T05:00:00.000Z}
  />
  <tbody>
    <ActivityRow
      activity={
        Object {
          "id": 1,
          "name": "Brush teeth",
          "value": 1,
        }
      }
      key="1"
      logsForActivity={
        Array [
          Object {
            "activityId": 1,
            "id": 1,
            "name": "Brush teeth",
            "occurredAt": 2018-04-08T05:00:00.000Z,
            "value": 1,
          },
          Object {
            "activityId": 1,
            "id": 2,
            "name": "Brush teeth",
            "occurredAt": 2018-04-08T05:00:00.000Z,
            "value": 1,
          },
        ]
      }
      weekDays={
        Array [
          2018-04-08T05:00:00.000Z,
          2018-04-09T05:00:00.000Z,
          2018-04-10T05:00:00.000Z,
          2018-04-11T05:00:00.000Z,
          2018-04-12T05:00:00.000Z,
          2018-04-13T05:00:00.000Z,
          2018-04-14T05:00:00.000Z,
        ]
      }
    />
    <ActivityRow
      activity={
        Object {
          "id": 2,
          "name": "Taekwondo",
          "value": 2,
        }
      }
      key="2"
      logsForActivity={
        Array [
          Object {
            "activityId": 2,
            "id": 3,
            "name": "Taekwondo",
            "occurredAt": 2018-04-11T05:00:00.000Z,
            "value": 2,
          },
        ]
      }
      weekDays={
        Array [
          2018-04-08T05:00:00.000Z,
          2018-04-09T05:00:00.000Z,
          2018-04-10T05:00:00.000Z,
          2018-04-11T05:00:00.000Z,
          2018-04-12T05:00:00.000Z,
          2018-04-13T05:00:00.000Z,
          2018-04-14T05:00:00.000Z,
        ]
      }
    />
    <ActivityRow
      activity={
        Object {
          "id": 3,
          "name": "Clean room",
          "value": 3,
        }
      }
      key="3"
      logsForActivity={
        Array [
          Object {
            "activityId": 3,
            "id": 4,
            "name": "Clean room",
            "occurredAt": 2018-04-10T05:00:00.000Z,
            "value": 3,
          },
        ]
      }
      weekDays={
        Array [
          2018-04-08T05:00:00.000Z,
          2018-04-09T05:00:00.000Z,
          2018-04-10T05:00:00.000Z,
          2018-04-11T05:00:00.000Z,
          2018-04-12T05:00:00.000Z,
          2018-04-13T05:00:00.000Z,
          2018-04-14T05:00:00.000Z,
        ]
      }
    />
  </tbody>
</table>
`;
